operators/go/gen_subm_operator.sh

```diff
@@ -61,23 +60,23 @@ function add_subm_engine_to_operator() {
   # Define spec fields
   types_file=pkg/apis/submariner/v1alpha1/submariner_types.go
   sed -i '/SubmarinerSpec struct/a \ \ Count int32 `json:"count"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerNamespace string `json:"submariner_namespace"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerClustercidr string `json:"submariner_clustercidr"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerServicecidr string `json:"submariner_servicecidr"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerToken string `json:"submariner_token"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerClusterid string `json:"submariner_clusterid"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerColorcodes string `json:"submariner_colorcodes"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerDebug string `json:"submariner_debug"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerNatenabled string `json:"submariner_natenabled"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ SubmarinerBroker string `json:"submariner_broker"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ BrokerK8sApiserver string `json:"broker_k8s_apiserver"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ BrokerK8sApiservertoken string `json:"broker_k8s_apiservertoken"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ BrokerK8sRemotenamespace string `json:"broker_k8s_remotenamespace"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ BrokerK8sCa string `json:"broker_k8s_ca"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ CeIpsecPsk string `json:"ce_ipsec_psk"`' $types_file
-  sed -i '/SubmarinerSpec struct/a \ \ CeIpsecDebug string `json:"ce_ipsec_debug"`' $types_file
-
-  # Define status fields, commented example:
+  sed -i '/SubmarinerSpec struct/a \ \ Namespace string `json:"namespace"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ ClusterCIDR string `json:"clusterCIDR"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ ServiceCIDR string `json:"serviceCIDR"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ Token string `json:"token"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ ClusterID string `json:"clusterID"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ ColorCodes string `json:"colorCodes"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ Debug string `json:"debug"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ NatEnabled string `json:"natEnabled"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ Broker string `json:"broker"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ BrokerK8sApiServer string `json:"brokerK8sApiServer"`' $types_file
+  sed -i '/SubmarinerSpec struct/a \ \ BrokerK8sApiServerToken string `json:"brokerK8sApiServerToken"`' $types_file
```

Comment:
About token and BrokerK8sApiServerToken I believe token is probably unnecessary, we need to investigate as I believe they are ultimately the same thing...

Reply:
Heh, I just happen to be looking into that too, trying to document the CR entries for the OLM package docs...

Reply:
> About token and BrokerK8sApiServerToken I believe token is probably unnecessary, we need to investigate as I believe they are ultimately the same thing...

Right, they both contain the same value so we should only need one. The Helm charts only use `broker.token`, which corresponds to `SUBMARINER_BROKER_TOKEN`, which corresponds to `SubmarinerToken`. I think we should keep `BrokerK8sApiServerToken` since it matches the other values and is Kubernetes-specific (isn’t it?).

Reply:
lol :)

Reply:
I agree, let's keep the BrokerK8sApiServerToken and remove the other, @dfarrell07  ^

Reply:
As discussed with @mangelajo on the upstream call today, I had a TODO to look into exactly this (do these tokens all make sense). Thanks for digging in, :+1: to keeping `BrokerK8sApiServerToken` and removing the duplicate.

Reply:
I raised a Jira to track this TODO here: https://jira.coreos.com/browse/HYCLD-274